src/main/java/me/mrCookieSlime/Slimefun/api/item_transport/CargoManager.java

```diff
@@ -1,228 +1,259 @@
+        if (!matchesFilter(node, stack, index)) return stack;
+
+        DirtyChestMenu menu = getChestMenu(target);
+
+        if (menu != null) {
+            for (int slot : menu.getPreset().getSlotsAccessedByItemTransport(menu, ItemTransportFlow.INSERT, stack)) {
+                ItemStack is = menu.getItemInSlot(slot) == null ? null : menu.getItemInSlot(slot).clone();
+
+                if (is == null) {
+                    menu.replaceExistingItem(slot, stack.clone());
+                    return null;
+                } else if (SlimefunManager.isItemSimilar(new CustomItem(is, 1), new CustomItem(stack, 1), true) && is.getAmount() < is.getType().getMaxStackSize()) {
+                    int amount = is.getAmount() + stack.getAmount();
+
+                    if (amount > is.getType().getMaxStackSize()) {
+                        is.setAmount(is.getType().getMaxStackSize());
+                        stack.setAmount(amount - is.getType().getMaxStackSize());
+                    } else {
+                        is.setAmount(amount);
+                        stack = null;
+                    }
+
+                    menu.replaceExistingItem(slot, is);
+                    return stack;
+                }
+            }
+        } else {
+            BlockState state = target.getState();
+
+            if (state instanceof InventoryHolder) {
+                Inventory inv = ((InventoryHolder) state).getInventory();
+
+                int minSlot = 0;
+                int maxSlot = inv.getContents().length;
+
+                //Check if it is a normal furnace
+                if (inv instanceof FurnaceInventory) {
+                    //Check if it is fuel or not
+                    if (stack.getType().isFuel()) {
+                        minSlot = 1;
+                        maxSlot = 2;
+                    } else {
+                        maxSlot = 1;
+                    }
+                } else if (inv instanceof BrewerInventory) {
+                    //Check if it goes in the potion slot,
+                    if (stack.getType() == Material.POTION || stack.getType() == Material.LINGERING_POTION || stack.getType() == Material.SPLASH_POTION) {
+                        maxSlot = 3;
+                        //The blaze powder slot,
+                    } else if (stack.getType() == Material.BLAZE_POWDER) {
+                        minSlot = 4;
+                        maxSlot = 5;
+                    } else {
+                        //Or the input
+                        minSlot = 3;
+                        maxSlot = 4;
+                    }
+                }
+
+                for (int slot = minSlot; slot < maxSlot; slot++) {
+                    ItemStack is = inv.getContents()[slot];
+
+                    if (is == null) {
+                        inv.setItem(slot, ChestManipulator.trigger(target, slot, null, stack.clone()));
+                        stack.setAmount(0);
+                        return stack;
```

Comment:
Why did you not return null there?

Reply:
I originally had a really bad system where I moved that to another method and that was the only way I could figure out if the item was placed fully or not placed at all. Should functionally be the same, but replaced it with a null anyways.